### PR TITLE
Link embedded rich text images to full size image

### PR DESCRIPTION
**Pull request by wakproductions**
### Context
See discussion: https://github.com/SpinaCMS/Spina/issues/1265

Would be nice if clicking on an embedded image takes the user to the original, full sized image. This would be useful for images with details that are best viewed at high resolution up close, or charts and tables that are difficult to read when downsized.

The proposed enhancement matches a similar feature found in Wordpress. Here is an example of a page where images are clickable to see the full sized image: https://winningwp.com/godaddy-vs-siteground-which-is-best-for-wordpress-hosting/

### Changes proposed in this pull request

Wraps images in an `<a>` tag with a link to the original image.

### Guidance to review

Up to @Bramjetten to decide whether this is a beneficial change.

### CODE REVIEW

**Comment by wakproductions** [comment]
I'm having trouble running the test suite locally. For some reason, when I run `rake test` it truncates the `schema_migrations` table and then thinks it has migrations to run. 🤷‍♂️

**Comment by Bramjetten** [comment]
Thanks for creating this PR! 

By default, the images Spina inserts are resized to 2000x2000, *not* the original size. You can override that by setting `embedded_image_size`.

File: app/assets/javascripts/spina/controllers/auto_file_upload_controller.js [hunk 1]
```diff
@@ -36,14 +36,4 @@ export default class extends Controller {
...
-  #imageData(imageMeta) {
-    return {
-      filename: imageMeta.dataset.filename,
-      signedBlobId: imageMeta.dataset.signedBlobId,
-      imageId: imageMeta.dataset.imageId,
-      embeddedUrl: imageMeta.dataset.embeddedUrl,
-      thumbnail: imageMeta.dataset.thumbnail
-    }
-  }
-
```
Comment (wakproductions) [inline app/assets/javascripts/spina/controllers/auto_file_upload_controller.js:48]: Accidentally forgot to delete this orphaned code in my last PR 🫢

File: app/assets/javascripts/spina/controllers/trix_controller.js [hunk 2]
```diff
@@ -33,9 +33,12 @@ export default class extends Controller {
...
   insertAttachment(event) {
-    let attachment = new Trix.Attachment({content: `<span class="trix-attachment-spina-image" data-label="Alt text">
-      <img src="${event.detail.embeddedUrl}" />
-    </span>`, contentType: "Spina::Image"})
+    const content =
+      `<span class="trix-attachment-spina-image" data-label="Alt text">
+       <a href="${event.detail.originalUrl}"><img src="${event.detail.embeddedUrl}" /></a>
+       </span>`
+
+    let attachment = new Trix.Attachment({content, contentType: "Spina::Image"})
```
Comment (wakproductions) [inline app/assets/javascripts/spina/controllers/trix_controller.js:41]: This just adds a link wrapper to images allowing the image to be clickable, leading to the full sized image.

**Comment by wakproductions** [comment]
Just added a small update to the code. Here's why this feature is useful. For things like stock charts or other kinds of data, the user might want the high rez version of the image to better see details.

spina-click-to-see-full-image-demo

**Comment by wakproductions** [comment]
> By default, the images Spina inserts are resized to 2000x2000, _not_ the original size. You can override that by setting `embedded_image_size`.

Thanks I made that correction in the docs addition here.

**Comment by wakproductions** [comment]
@Bramjetten if you don't like this feature, what do you think about modifying `RichTextPresenter` so that image type of data is passed off to another object, maybe named `Embeds::Image`. That object could be overridden in the main program to have whatever behavior the website creator wants for images? We can even create configuration settings in `initializers/spina.rb` to be accessed by that object so that rich text images can be handled however specified?

**Comment by MSchmidt** [comment]
IMHO I wouldn't want this as a default behaviour. Or if default at least configurable. I'd prefer to see images that can link to anything but spina's trix doesn't allow links around images right now.

**Comment by wakproductions** [comment]
> IMHO I wouldn't want this as a default behaviour. Or if default at least configurable. I'd prefer to see images that can link to anything but spina's trix doesn't allow links around images right now.

Understandable. I might try try a different approach with this then - making an image embed component that has configurable properties like "clickable" and "caption" that would have its own wrapper template.

**Pull request closed without merging**